### PR TITLE
Add option for looping backwards over parallel loops

### PR DESCRIPTION
**Pull request by adayton1**
* This may be useful for locating race conditions in a host build

### CODE REVIEW

File: src/care/forall.h [hunk 1]
```diff
@@ -71,14 +71,33 @@ namespace care {
...
+         RAJA::RangeStrideSegment rangeSegment;
+
+         if (std::is_same<ExecutionPolicy, openmp>::value ||
+             std::is_same<ExecutionPolicy, gpu>::value ||
+             std::is_same<ExecutionPolicy, parallel>::value ||
+             std::is_same<ExecutionPolicy, gpu_simulation>::value) {
+            rangeSegment = RAJA::RangeStrideSegment(end - 1, start - 1, -1);
+         }
+         else {
+            rangeSgement = RAJA::RangeStrideSegment(start, end, 1);
```
Comment (liu15) [inline src/care/forall.h:89]: Sgement -> Segment

File: src/care/forall.h [hunk 2]
```diff
@@ -71,14 +71,33 @@ namespace care {
       const int length = end - start;
 
       if (length != 0) {
+
 #ifndef CARE_DISABLE_RAJAPLUGIN
          RAJAPlugin::pre_forall_hook(ExecutionPolicyToSpace<ExecutionPolicy>::value, fileName, lineNumber);
 #endif
+
+#if CARE_ENABLE_PARALLEL_LOOP_BACKWARDS
+         RAJA::RangeStrideSegment rangeSegment(start, end, 1);
```
Comment (liu15) [inline src/care/forall.h:80]: No default constructor, eh?

Would it be better either define segmentStart, segmentEnd, and segmentStride as variables can call rangeSegment(segmentStart, segmentEnd, segmentStride)?  Or defined a boolean isParallel and initialize using a ternary operator?

Comment (adayton1) [reply src/care/forall.h]: I'll go with the ternary operator.

File: src/care/forall.h (hunk 1 above)
Comment (adayton1) [reply src/care/forall.h]: Fixed.

**Comment by adayton1** [comment]
OK, now this is working and ready for review.

File: src/care/forall.h [hunk 3]
```diff
@@ -193,6 +228,10 @@ namespace care {
 #else
```
Comment (dtaller) [inline src/care/forall.h:228]: This is totally unrelated to this PR, but I noticed we have a few lines above

forall(RAJA::hip_exec<CARE_CUDA_BLOCK_SIZE, CARE_CUDA_ASYNC>{},
             fileName, lineNumber, start, end, body);

above. In the future we should remember to either have CARE_GPU_BLOCK_SIZE (just rename it) or have separate CARE_CUDA_BLOCK_SIZE and CARE_HIP_BLOCK_SIZE

**Review by dtaller — APPROVED** [review]
Looks good to me

File: src/care/forall.h (hunk 3 above)
Comment (adayton1) [reply src/care/forall.h:228]: Yeah, we'll need to clean that up. Thanks for pointing that out.

Comment (adayton1) [reply src/care/forall.h:228]: I created an issue so that we remember to fix this.

**Review by liu15 — APPROVED** [review]

**Pull request merged**